log-elasticsearch/src/Log/Backend/ElasticSearch/V5.hs

```diff
@@ -144,19 +143,29 @@ elasticSearchLogger esConf@ElasticSearchConfig{..} genRandomWord = do
           $ "ElasticSearch: unexpected response: " <> jsonToBSL replyBody
         Just (hasErrors, items) -> when hasErrors $ do
           -- If any message failed to be inserted because of type mismatch, go
-          -- back to them, replace their data with elastic search error and put
-          -- old data into its own namespace to work around insertion errors.
+          -- back to them, log the insertion failure and add type suffix to each
+          -- of the keys in their "data" fields to work around type errors.
           let failed = V.findIndices (H.member "error") items
-          dummyMsgs <- V.forM failed $ \n -> do
-            dataNamespace <- liftIO genRandomWord
-            let modifyData oldData = object [
-                    "__es_error" .= H.lookup "error" (items V.! n)
-                  , "__es_modified" .= True
-                  , ("__data_" <> showt dataNamespace) .= oldData
-                  ]
-            return . second (H.adjust modifyData "data") $ jsonMsgs V.! n
+              newMsgs = (`V.map` failed) $ \n ->
+                let modifyData :: Value -> Value
+                    modifyData (Object hm) = Object $ H.fromList
+                      [ "__es_error" .= H.lookup "error" (items V.! n)
+                      , "__es_modified" .= True
+                      ] `H.union` H.foldlWithKey' keyAddValueTypeSuffix H.empty hm
+                    modifyData v = v
```

Comment:
This case shouldn't occur, right? Because `data` value is always an object. Or could it be something else?

Reply:
It could be anything as `log*` functions accept any type with `ToJSON` instance as additional data.